### ISSUE TITLE
DDWSGIMiddleware - 'NoneType' object has no attribute 'set_tag' when making requests between python services

### ISSUE THREAD

**Issue by alexkasemir** [DataDog/dd-trace-py#2586]
### Which version of dd-trace-py are you using?
0.49.3

### Which version of pip are you using?

The latest

### Which version of the libraries are you using?

```
attrs==19.2.0
bottle==0.12.13
ddtrace==0.49.3
requests==2.21.0
sentry-sdk==0.20.3
uWSGI==2.0.18
wsgi-request-logger==0.4.6
```
(the relevant libraries)

### How can we reproduce your problem?
We deploy our bottle api using `uwsgi` and we have followed the installation instructions [here](https://ddtrace.readthedocs.io/en/stable/advanced_usage.html#uwsgi).

Instead of the bottle plugin for tracing we are using the [WSGI middleware](https://ddtrace.readthedocs.io/en/stable/integrations.html#wsgi). The reason for this, is we want to trace the wsgi.request module. Particularly, we use the [wsgi-request-logger](https://github.com/pklaus/wsgi-request-logger) to log our requests. Using the DDWSGIMiddleware, these log lines are appropriately traced.

However, when making a request from one service to another and then back to the original, we hit an exception in the ddtrace-py source code. The error occurs when the original service is returning back to the second service.

**service1** _--- (makes a requests using requests module )--->_ **service2** _--- (makes a requests using requests module Error occurs here)--->_ **service1**

service2 is another python application instrumented with datadog.

### What is the result that you get?
[This line](https://github.com/DataDog/dd-trace-py/blob/bb126e86d79655ec5471a284ad95bb10cc36ba31/ddtrace/contrib/wsgi/wsgi.py#L101) errors out and we cannot finish the request. This means that a root span could not be found. I am not sure why this is the case.

### What is the result that you expected?
Even if a root span could not be found, ddtrace should not blow up. (also I do not know why a root span could not be found)

Any advice? At the very least it seems like the `DDWSGIMiddleware` code should be null safe.

**Comment by Kyle-Verhoog** [comment]
Hey @alexkasemir! Thanks for the details report and sorry for the delay! Hmm really strange that there's no root span in the context in which the `start_response` is occurring... but it could be due to some funky usage which we didn't anticipate.

I've opened up a fix in #2659.

I'd love to get a repro of your scenario just so we can have a regression test to avoid this failure in the future 🙂 Is this error occurring with the wsgi-request-logger? Or in some other wsgi middleware?

**Comment by deveshvar** [comment]
Same issue.
`ddtrace==0.50.0`
I am setting up fresh Datadog APM in 2 django based services.
**Service1** (gUnicorn running Uvicorn worker, hence using ASGI **TraceMiddleware**)
**Service2** (gUnicorn running WSGI application, hence using **DDWSGIMiddleware**)

API calls from Service1 -to-> Service2 result in the **exception**. (requests module)
API calls from Postman/Browser -to> Service2 work fine.
Also, API calls from Service2 -to-> Service1 work fine. (requests module)

* Not using `wsgi-request-logger`

**Comment by alexkasemir** [comment]
@Kyle-Verhoog The error is being thrown in `DDWSGIMiddleware` and occurs even if I strip out all other middlewares from our application.

**Comment by Kyle-Verhoog** [comment]
@alexkasemir gotcha, thanks! The fix has been released as 0.50.1 which should be up on pypi shortly!
